indexing-service/src/main/java/io/druid/indexing/firehose/IngestSegmentFirehoseFactory.java

```diff
@@ -164,12 +165,20 @@ public Firehose connect(InputRowParser inputRowParser) throws IOException, Parse
       List<String> dims;
       if (dimensions != null) {
         dims = dimensions;
+      } else if (inputRowParser.getParseSpec().getDimensionsSpec().hasCustomDimensions()) {
+        dims = inputRowParser.getParseSpec().getDimensionsSpec().getDimensions();
```

Comment:
Anything that uses map input row parser uses dimension exclusions. Hadoop indexing for example uses dimension exclusions.